packages/components/src/Button/DirectionalLink/DirectionalLink.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react"
+import {
+  ArrowBackwardIcon,
+  ArrowForwardIcon,
+  EndIcon,
+  StartIcon,
+} from "~components/Icons"
+import { GenericButton, GenericProps } from "../GenericButton"
+
+export interface DirectionalLinkProps extends GenericProps {
+  label: string
+  disabled?: boolean
+  direction: "prev" | "next" | "start" | "end"
+}
```

Comment:
I'd probably rename `GenericProps` to `GenericButtonProps` too.

```suggestion
export type DirectionalLinkProps = {
  label: string
  disabled?: boolean
  direction: "prev" | "next" | "start" | "end"
} & GenericProps
```